### PR TITLE
Nitishsharma/mysql connector changes

### PR DESCRIPTION
**Pull request by nitishsharma25**
1. Replaced pymysql connector with mysql connector.

- Under connection parameters, read/write timeout and bindAddress params are not supported in mysql connector.
- Removed cursor.mogrify statements as it is not supported in mysql connector
- Added buffered=True since buffered=False throws an error.
- use_pure flag is set to False by default (Use the c implementation first) so did not set it explicitly.
 
2. Added AAD support Added with auth_plugin parameter in the connection if azureAccountToken param is provided.

3. Modified some converter's function and removed which are not used or needed.

4. Added and fixed UT's for connection and driver.

5. Changed file names and syntaxes from pymysql to mysql

### CODE REVIEW

File: ossdbtoolsservice/converters/converters.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
 
 def convert_str(value: str):
-    return bytearray(value.encode())
+    return bytearray(str(value).encode())
```
Comment (rishky-msft) [inline ossdbtoolsservice/converters/converters.py:24]: passed argument is already str. Why we have done this?

File: ossdbtoolsservice/driver/types/mysql_driver.py [hunk 2]
```diff
@@ -21,18 +21,17 @@
...
-    'bindAddress': 'bind_address',
-    'readTimeout': 'read_timeout',
-    'writeTimeout': 'write_timeout',
+    'connectTimeout': 'connection_timeout',
     'sqlMode': 'sql_mode',
-    'clientFlag': 'client_flag'
+    'clientFlag': 'client_flags',
+    'ssl.key': 'ssl_key',
+    'ssl.ca': 'ssl_ca',
+    'ssl.cert': 'ssl_cert'
```
Comment (rishky-msft) [inline ossdbtoolsservice/driver/types/mysql_driver.py:29]: Check if cipher can also be included in params.

File: ossdbtoolsservice/driver/types/mysql_driver.py [hunk 3]
```diff
@@ -21,18 +21,17 @@
 
 MYSQL_CONNECTION_OPTION_KEY_MAP = {
     'dbname': 'database',
-    'connectTimeout': 'connect_timeout',
-    'bindAddress': 'bind_address',
-    'readTimeout': 'read_timeout',
```
Comment (rishky-msft) [inline ossdbtoolsservice/driver/types/mysql_driver.py:26]: Check if read and write timeouts can be added in mysql connector.

File: ossdbtoolsservice/query/column_info.py [hunk 4]
```diff
@@ -12,10 +12,6 @@ def get_columns_info(cursor) -> List[DbColumn]:
     if cursor.description is None:
         raise ValueError('Cursor description is not available')
 
-    if cursor.connection is None:
```
Comment (rishky-msft) [inline ossdbtoolsservice/query/column_info.py:15]: Why this is removed? What is the usecase of this?

File: ossdbtoolsservice/driver/types/mysql_driver.py [hunk 5]
```diff
@@ -229,7 +232,7 @@ def cursor(self, **kwargs):
         """
         self._conn.ping()
         # Create a new cursor from the current connection
-        cursor_instance = self._conn.cursor()
+        cursor_instance = self._conn.cursor(buffered=True)
```
Comment (rishky-msft) [inline ossdbtoolsservice/driver/types/mysql_driver.py:235]: Can we keep buffered = False?
Do it improves the performance for large read queries?

File: ossdbtoolsservice/driver/types/mysql_driver.py [hunk 6]
```diff
@@ -95,21 +100,20 @@ def __init__(self, conn_params: {}, config: Optional[Configuration] = None):
         if 'port' not in self._connection_options or not self._connection_options['port']:
             self._connection_options['port'] = constants.DEFAULT_PORT[constants.MYSQL_PROVIDER_NAME]
 
-        self._set_ssl_options(conn_params)
-
         # Setting autocommit to True initally
         self._autocommit_status = True
```
Comment (rishky-msft) [inline ossdbtoolsservice/driver/types/mysql_driver.py:104]: If we are not setting usePure explicitly, 
Can we add a log/metric to capture fallbacks?

File: ossdbtoolsservice/converters/converters.py (hunk 1 above)
Comment (nitishsharma25) [reply ossdbtoolsservice/converters/converters.py]: Removed it

File: ossdbtoolsservice/driver/types/mysql_driver.py (hunk 6 above)
Comment (nitishsharma25) [reply ossdbtoolsservice/driver/types/mysql_driver.py:107]: Added check and logging for which extension the connector is using.

File: ossdbtoolsservice/driver/types/mysql_driver.py (hunk 5 above)
Comment (nitishsharma25) [reply ossdbtoolsservice/driver/types/mysql_driver.py:244]: It does not improve perf, at best it is same

File: ossdbtoolsservice/driver/types/mysql_driver.py (hunk 2 above)
Comment (rishky-msft) [reply ossdbtoolsservice/driver/types/mysql_driver.py:30]: No option for cipher in documentation.

File: ossdbtoolsservice/driver/types/mysql_driver.py (hunk 3 above)
Comment (rishky-msft) [reply ossdbtoolsservice/driver/types/mysql_driver.py:26]: No option for read and write timeout in documentation.

File: ossdbtoolsservice/query/column_info.py (hunk 4 above)
Comment (rishky-msft) [reply ossdbtoolsservice/query/column_info.py:15]: this code piece is not being used or hit. We can remove it.

**Comment by rishky-msft** [comment]
Remove the cipher, readtimeout and writetimeout params from pgtoolsservice and azuredatastudio-mysql.

Pgtoolsservice : https://github.com/microsoft/pgtoolsservice/blob/nitishsharma/mysql-connector-changes/ossdbtoolsservice/capabilities/connection_options/mysql_connection_options.py

azuredatastudio-mysql:  https://github.com/microsoft/azuredatastudio-mysql/blob/main/package.json

**Review by rishky-msft — APPROVED** [review]

**Review by rishky-msft — APPROVED** [review]

**Pull request merged**